lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -589,6 +618,33 @@ static LogicalResult verifyEnableOp(EnableOp enableOp) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// IfOp
+//===----------------------------------------------------------------------===//
+static LogicalResult verifyIfOp(IfOp ifOp) {
+  auto component = ifOp->getParentOfType<ComponentOp>();
+  auto wiresOp = component.getWiresOp();
+  auto groupName = ifOp.groupName();
+  auto groupOp = wiresOp.lookupSymbol<GroupOp>(groupName);
+
+  if (!groupOp)
+    return ifOp.emitOpError()
+           << "with group '" << groupName << "', which does not exist.";
+
+  if (ifOp.thenRegion().front().empty())
+    return ifOp.emitError() << "empty 'then' region.";
+
+  if (ifOp.elseRegion().getBlocks().size() != 0 &&
+      ifOp.elseRegion().front().empty())
+    return ifOp.emitError() << "empty 'else' region.";
+
+  if (failed(isPortDrivenByGroup(ifOp.cond(), groupOp)))
+    return ifOp.emitError() << "conditional port expected to be driven from "
```

Comment:
tiny nit: `"conditional port: " << ifOp.cond() << " expected ..."`

Reply:
I've been a bit frustrated with not having a "clean" api for printing SSA values. Added the `valueName` function in the most recent commit to get this name - let me know what you think.